crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -417,20 +415,41 @@ where
         trace!(target: "rpc::eth::filter", from=from_block, to=to_block, ?filter, "finding logs in range");
 
         if to_block - from_block > self.max_blocks_per_filter {
-            return Err(FilterError::QueryExceedsMaxBlocks(self.max_blocks_per_filter))
+            return Err(FilterError::QueryExceedsMaxBlocks(self.max_blocks_per_filter));
         }
 
+
         let mut all_logs = Vec::new();
         let filter_params = FilteredParams::new(Some(filter.clone()));
 
         // derive bloom filters from filter input
         let address_filter = FilteredParams::address_filter(&filter.address);
         let topics_filter = FilteredParams::topics_filter(&filter.topics);
 
+        if (to_block == from_block) && (from_block == self.provider.last_block_number()?) {
+            let mut all_logs = Vec::new();
+            // let x = self.provider.convert_block_number(BlockNumberOrTag::from(to_block));
+            let x = self.provider.block_hash(to_block);
+            if let Some((block, receipts)) =
+                self.eth_cache.get_block_and_receipts(x.unwrap().unwrap()).await?
+            {
+                let b: BlockNumHash = <BlockNumHash as From<(u64, revm_primitives::FixedBytes<32>)>>
+                ::from((to_block, block.hash));
+                logs_utils::append_matching_block_logs(
+                    &mut all_logs,
+                    &filter_params,
+                    b,
+                    block.body.into_iter().map(|tx| tx.hash()).zip(receipts),
+                    false,
+                );
+            }
+            return Ok(all_logs);
```

Comment:
Hey! 
small nit: I think that the idiomatic way in Rust and across the repo is not to add ";" in return statements 
maybe you have some auto-fmt tooling that put ";" after returns
I got such problems a few times in my rust setup
Try to use "cargo +nightly fmt" , it helps!
ps
cargo clippy --bin "reth" --workspace --features "ethereum" --lib --tests --benches --examples
and this one to check lints